### PR TITLE
Added editorconfig. Added tests.

### PR DESCRIPTION
**Pull request by andreasrs**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,8 @@
   "description": "Thunk middleware for Redux.",
   "main": "lib/index.js",
   "scripts": {
-    "prepublish": "rm -rf lib && babel src --out-dir lib"
+    "prepublish": "rm -rf lib && babel src --out-dir lib",
```
Comment (gaearon) [inline package.json]: Let's use `rimraf` like [here](https://github.com/gaearon/redux/blob/a467c4174ecd99e335b5160c1ce163141d438844/package.json#L7-L21) so it builds on Windows too.

**Comment by gaearon** [comment]
Thanks! Some more things worth checking:

1. If action isn't a function, the return value is whatever `next` returned.
2. If action is a function, the return value is the return value of that function.
3. If action is a function, it is invoked synchronously.

**Comment by andreasrs** [comment]
Sure!

I have some spare time this weekend so I'll get this pushed.

**Comment by andreasrs** [comment]
The last two commits here should take care of those three points. 

Not completely sure if the explicit check for synchronicity is required (checked implicitly by the other test?), but I guess it serves as documentation.

**Comment by gaearon** [comment]
Great job.

**Pull request merged**